sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnInvoker.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.transforms;
+
+/** Interface for invoking the {@code DoFn} processing methods. */
```

Comment:
Mention that:
`Instantiating a DoFnInvoker associates it with a specific {@link DoFn} instance, referred to as the bound {@link DoFn}.`

It just now occurs to me I could have made this take `DoFn<InputT, OutputT>` as well, which would have let us reuse `DoFnInvoker` instances (rather than just constructors). Something we could consider in a followup.